arch/posix/CMakeLists.txt

```diff
@@ -1,12 +1,17 @@
 zephyr_compile_options(
-  -fno-freestanding
   -m32
   -MMD
   -MP
   ${ARCH_FLAG}
   -include ${ZEPHYR_BASE}/arch/posix/include/posix_cheats.h
   )
 
+if(NOT CMAKE_C_COMPILER_ID STREQUAL "Clang") # clang errors on -fno-freestanding
+zephyr_compile_options(
```

Comment:
We can and should avoid this if statement by doing zephyr_cc_options() instead of zephyr_compile_options.